### PR TITLE
label propagate

### PR DESCRIPTION
**Pull request by SoluMilken**
Propagate labels according to given SpanGroup or EditGroup.

### CODE REVIEW

File: uttut/pipeline/bian/label_propagation.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+    # final
+    output_labels[o_start:] = labels[i_start:]
+    return output_labels
+
+
+def _get_max_label(labels: List[int]):
+    if len(labels) == 0:
+        return 0
+    return max(labels)
```
[the remark this answers is not in the document]
Comment (SoluMilken) [reply uttut/pipeline/bian/label_propagation.py]: 好喔好喔

[the remark this answers is not in the document]
Comment (SoluMilken) [reply uttut/pipeline/bian/label_propagation.py]: d8ecbde

File: uttut/pipeline/bian/label_propagation.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+            span_group=span_group,
+            reduce_func=reduce_func,
+        )
+    elif len(span_group) == len(labels):
+        return _untokenize(
+            labels=labels,
+            span_group=span_group,
+        )
+    else:
+        raise KeyError('labels and span_group are not competible.')
```
Comment (absolutelyNoWarranty) [inline uttut/pipeline/bian/label_propagation.py]:
```suggestion
        raise KeyError('labels and span_group are not compatible.')
```

Comment (SoluMilken) [reply uttut/pipeline/bian/label_propagation.py]: spelling error :scream:

File: uttut/pipeline/bian/label_propagation.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        reduce_func: Callable[[List[int]], int] = None,
+    ) -> List[int]:
+    '''Map the labels[fstart_i: fend_i] to output_labels[rstart_i: rend_i]
+    Note that the length of edit_group should be the same as that of backward_edit_group.
+    Map the reduced label in [forward_edit[i].start, forward_edit[i].end)
+    to output list
+    Args:
+        labels (list of ints)
+        edit_group (EditGroup)
+        reverse_edit_group (EditGroup)
```
[the remark this answers is not in the document]
Comment (SoluMilken) [reply uttut/pipeline/bian/label_propagation.py]: :scream:

File: uttut/pipeline/bian/label_propagation.py (hunk 2 above)
Comment (SoluMilken) [reply uttut/pipeline/bian/label_propagation.py]: b0eab78

File: uttut/pipeline/bian/label_propagation.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (SoluMilken) [reply uttut/pipeline/bian/label_propagation.py]: b0eab78

File: uttut/pipeline/bian/label_propagation.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+        o_start += expand_size
+
+    # final
+    output_labels[o_start:] = labels[i_start:]
+    return output_labels
+
+
+def _get_most_common_label(labels: List[int], output_size: int = 1):
+    if len(labels) == 0:
+        most_common_label = 0
```
Comment (noobOriented) [inline uttut/pipeline/bian/label_propagation.py]: why 0?

**Comment by SoluMilken** [comment]
@GBLin5566

**Pull request merged**